pkg/api/pod/util.go

```diff
@@ -324,31 +324,17 @@ func dropDisabledRunAsGroupField(podSpec, oldPodSpec *api.PodSpec) {
 }
 
 // dropDisabledProcMountField removes disabled fields from PodSpec related
-// to ProcMount
+// to ProcMount only if it is not already used by the old spec
 func dropDisabledProcMountField(podSpec, oldPodSpec *api.PodSpec) {
-	if !utilfeature.DefaultFeatureGate.Enabled(features.ProcMountType) {
-		defProcMount := api.DefaultProcMount
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ProcMountType) && !procMountInUse(oldPodSpec) {
 		for i := range podSpec.Containers {
 			if podSpec.Containers[i].SecurityContext != nil {
-				podSpec.Containers[i].SecurityContext.ProcMount = &defProcMount
```

Comment:
populating this field when the feature was disabled was a mistake that wasn't caught during review of https://github.com/kubernetes/kubernetes/pull/64283. it should have been set to nil here, but was not.

unfortunately, that means we have existing objects with this field set to `Default`, despite it being alpha level. preserving that behavior is the lowest risk thing we can do at this point.

I would recommend the following:
1. move the `nil -> &api.DefaultProcMount` defaulting to be an actual defaulting rule in `pkg/apis/core/v1/defaults.go` (rather than this odd feature-gated defaulting):

    ```go
    func SetDefaults_SecurityContext(obj *v1.SecurityContext) {
      if obj.ProcMount == nil {
        defProcMount := v1.DefaultProcMount
        obj.ProcMount = &defProcMount
      }
    }
    ```

2. make `procMountInUse` return true if any procMount value is non-nil and is not set to the api.DefaultProcMount (if all the procmount fields are nil or the default, we do not consider the feature in use)

3. preserve the behavior in dropDisabledProcMountField of setting to `&defProcMount`

Reply:
Done